cou/logging.py

```diff
@@ -71,6 +71,26 @@ def setup_logging(log_level: str = "INFO") -> None:
     # suppress stack trace on console
     console_handler.addFilter(TracebackInfoFilter())
 
+    # suppress python libjuju debug logs
+    if log_level != "NOTSET":
+        log_file_handler.addFilter(filter_debug_logs)
+
     root_logger.addHandler(log_file_handler)
     root_logger.addHandler(console_handler)
     logger.info("Logs of this execution can be found at %s", file_name)
+
+
+def filter_debug_logs(record: logging.LogRecord) -> bool:
+    """Filter debug logs to not go to the logfile.
+
+    libjuju and websockets are very verbose on the debug mode and the logfile
+    can be huge if not filtered.
+    :param record: A LogRecord instance represents an event being logged.
+    :type record: LogRecord
+    :return: Returns false to not append record in the log file, true for appending it.
+    :rtype: bool
+    """
+    return not (
+        (record.name.startswith("juju.") or record.name.startswith("websockets."))
+        and record.levelname == "DEBUG"
+    )
```

Comment:
I think it will be effective to check first if record.levelname is DEBUG to avoid doing `startswith` twice for no debug messages.
```suggestion
    return (
        record.levelname != "DEBUG"
        or not (record.name.startswith("juju.") or record.name.startswith("websockets."))
    )
```